src/main/java/com/invoiced/entity/AbstractEntity.java

```diff
@@ -12,9 +12,29 @@
 
 public abstract class AbstractEntity<T extends AbstractEntity> {
 
-	protected Connection conn;
+	private Connection conn;
 	protected Class<T> tClass;
-	private boolean entityCreated;
+	protected boolean entityCreated;
+	protected String entityName;
+	private String endpointBase = "";
+
+	protected void setEndpointBase(String base) {
+		this.endpointBase = base;
+	}
+
+	protected String getEndpoint(boolean includeId, boolean includeBaseUrl) {
+		String url = "";
+
+		if (includeBaseUrl) url += this.conn.baseUrl();
```

Comment:
I understand but would lean toward letting getEndpoint fetch the base url from the connection class, because it is already called for all request methods. The alternative inside of AbstractEntity would be slightly more verbose but not terribly (e.g. in retrieve for example, the url line would go from

`String url = this.getEndpoint(false, true) + "/" + id;`
to something like
`String url = this.conn.baseUrl() + this.getEndpoint(false, false) + "/" + id;`

And it would be necessary to add a getter for the base url that the subclasses could use. Thoughts?